src/main/java/org/jabref/logic/importer/fileformat/EndnoteXmlImporter.java

```diff
@@ -0,0 +1,212 @@
+package org.jabref.logic.importer.fileformat;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Unmarshaller;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamReader;
+
+import org.jabref.logic.importer.ImportFormatPreferences;
+import org.jabref.logic.importer.Importer;
+import org.jabref.logic.importer.ParseException;
+import org.jabref.logic.importer.Parser;
+import org.jabref.logic.importer.ParserResult;
+import org.jabref.logic.importer.fileformat.endnote.Keywords;
+import org.jabref.logic.importer.fileformat.endnote.PdfUrls;
+import org.jabref.logic.importer.fileformat.endnote.Record;
+import org.jabref.logic.importer.fileformat.endnote.Xml;
+import org.jabref.logic.util.FileType;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.BiblatexEntryTypes;
+import org.jabref.model.entry.FieldName;
+import org.jabref.model.entry.LinkedFile;
+import org.jabref.model.strings.StringUtil;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Importer for the Endnote XML format.
+ *
+ * Based on dtd scheme downloaded from Article #122577 in http://kbportal.thomson.com.
+ */
+public class EndnoteXmlImporter extends Importer implements Parser {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EndnoteXmlImporter.class);
+    private final ImportFormatPreferences preferences;
+
+    public EndnoteXmlImporter(ImportFormatPreferences preferences) {
+        this.preferences = preferences;
+    }
+
+    @Override
+    public String getName() {
+        return "EndNote XML";
+    }
+
+    @Override
+    public FileType getFileType() {
+        return FileType.ENDNOTE_XML;
+    }
+
+    @Override
+    public String getId() {
+        return "endnote";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Importer for the EndNote XML format.";
+    }
+
+    @Override
+    public boolean isRecognizedFormat(BufferedReader reader) throws IOException {
+        String str;
+        int i = 0;
+        while (((str = reader.readLine()) != null) && (i < 50)) {
+            if (str.toLowerCase(Locale.ENGLISH).contains("<records>")) {
+                return true;
+            }
+
+            i++;
+        }
+        return false;
+    }
+
+    @Override
+    public ParserResult importDatabase(BufferedReader reader) throws IOException {
+        Objects.requireNonNull(reader);
+
+        try {
+            JAXBContext context = JAXBContext.newInstance("org.jabref.logic.importer.fileformat.endnote");
```

Comment:
Constructing a `JAXBContext` is very expensive. It is so expensive that you notice it even as a human. Please turn this into an instance variable, then the construction takes place only once per importer object. Or you could just store the `Unmarshaller`, which should be enough as well.

Reply:
Good to know! I actually copied this part from the medline importer and will change it there too.